### PR TITLE
Feature osmp dummy function

### PR DESCRIPTION
**Pull request by albinnyASTAzero**
Added dummy function (simple AEB functionality). **Please focus on evaluateAEB (row 292 in .cpp) and doCalc (row 331 in .cpp) functions, everything else is copy-paste from OSI main repo.** 

- Used OSMP FMI/FMU example and created OSMPDummyFunction that consumes osi.SensorData messages and output osi.TrafficUpdate. 
- doCalc is main FMU function that is called every step from simulation. It reads the necessary info from SensorData, evaluates if AEB should be applied and outputs TrafficUpdate with decelRequest.  
- Made small addition in OSMPDummySensor to also send relative velocity of target.

### CODE REVIEW

File: examples/OSMPDummyFunction/OSMPDummyFunction.cpp [hunk 1]
```diff
@@ -0,0 +1,878 @@
...
+
+    double pbTTClimit = 2.5; // prebrake 
+    double pbRequest = -3.0;
+
+    double fbTTClimit = 1.5; // fullbrake
+    double fbRequest = -5.0; 
+
+    // TODO add information to Sensordata so that this could be derived
+    double collisionWidth = 1.5; // ~Width/2 of target (Could be derived with getting dimension of target) 
+    double distanceOffset = 4.5; // ~Length/2 of target + ~length/2 of ego. (Since Sensorview calculates distance from base of ego to base of target)
```
Comment (oscjoha) [inline examples/OSMPDummyFunction/OSMPDummyFunction.cpp]: All these as class members I think is a good idea.

**Review by oscjoha — CHANGES_REQUESTED** [review]
A lot of things we don't really need to care about here. Would recommend inheriting from the base OSI dummy sensor class and implement your functions in that instead.

File: examples/OSMPDummyFunction/OSMPDummyFunction.cpp (hunk 1 above)
Comment (albinnyASTAzero) [reply examples/OSMPDummyFunction/OSMPDummyFunction.cpp]: Ok. What is the best way to initialize them?

**Comment by albinnyASTAzero** [comment]
> A lot of things we don't really need to care about here. Would recommend inheriting from the base OSI dummy sensor class and implement your functions in that instead.

Yeah I agree. Problem is that they have different I/O and the current I/O getters/setter functions and variables of dummysensor are named after the I/O which could be a bit misleading. 
Ideally, we should make a general OSMP class that both dummySensor and function could inherit from. So let's have that as an todo for the future :)

File: examples/OSMPDummyFunction/OSMPDummyFunction.cpp (hunk 1 above)
Comment (albinnyASTAzero) [reply examples/OSMPDummyFunction/OSMPDummyFunction.cpp]: Directly in the header feels like the simplest option but is there a better way ?

Comment (albinnyASTAzero) [reply examples/OSMPDummyFunction/OSMPDummyFunction.cpp]: I put in header file for now. Please tell me if you think this is incorrect.

Comment (oscjoha) [reply examples/OSMPDummyFunction/OSMPDummyFunction.cpp]: Nothing wrong in doing it like that in my opinion

**Comment by oscjoha** [comment]
> 

What I/O you mean? If there are a few methods called in your doCalc that have confusing name just make a more reasonable name and call the base class method in that one, like a wrapper.

**Comment by albinnyASTAzero** [comment]
> > 
> 
> What I/O you mean? If there are a few methods called in your doCalc that have confusing name just make a more reasonable name and call the base class method in that one, like a wrapper.

DummySensor has I/O, SensorView/SensorData, while DummyFunction has SensorData/TrafficUpdate. So they have named all the variables and getter/setter functions to that (f.ex get_fmi_sensorview_in/set_fmi_sensordata_out). Also the buss definitions at the top of the header files are named according to that. 

So as I said, it could be a nice idea to make a general OSMP class that has general I/O, that which all can inherit from rather than OSMPFunction inheriting from Sensor I think.

**Comment by albinnyASTAzero** [comment]
Merged. Closing

**Pull request merged**